### PR TITLE
refinements: removed from individual cases

### LINKED ISSUE #645: generator: refinements and lexical scope.
context: https://github.com/exercism/xruby/pull/583#pullrequestreview-37758023

Make the underscoring helper methods available to <problem>_cases without them having to remember to include 'using Generator::Underscore'

(This may involve not using refinements.)

### PR DESCRIPTION
**Pull request by ajwann**
fixes #645

### CODE REVIEW

File: lib/generator/exercise_case.rb [hunk 1]
```diff
@@ -27,5 +27,16 @@ def method_missing(sym, *args, &block)
     def respond_to?(sym, include_private = false)
       canonical.respond_to?(sym) || super
     end
+
+    protected
+
+    def underscore(target)
+      target.underscore
```
Comment (ajwann) [inline lib/generator/exercise_case.rb]: I think it's best if we still use the refinement within ```ExerciseCase```. If we don't, we will have to define two separate ```underscore``` methods, one for ```Fixnum``` and one for ```String```, or define a single ```underscore``` method that switches on the type of the argument, but I really dislike that option.

Comment (Insti) [inline lib/generator/exercise_case.rb]: I agree that switching on type is undesirable.

**Comment by Insti** [comment]
Getting rid of the refinements in the <problem>Case classes is good 👍 and I think what you have here is mergable.

I'm not sure continuing to use the function with refinements the way we are here is a good idea. Underscoring a string and underscoring an Integer are two fundamentally different operations performed for different reasons, yet they are called using the same function.

Perhaps two different methods is the right way to go, but naming them based on `why` rather than `what`.

@kotp had some thoughts about this (that I can't find to link to right now.)

**Comment by ajwann** [comment]
I agree that string underscore and integer underscore are very different operations, and I'm in favor of creating the ```literal(number)``` and ```underscore(string)``` methods.

**Comment by Insti** [comment]
>  I'm in favor of creating the `literal(number)` and `underscore(string)` methods.

Is this something you want to do as part of this PR?

**Comment by ajwann** [comment]
@Insti sure, no problem! I can make the change right now actually.

**Comment by ajwann** [comment]
@Insti I think this is  good to, but let me know if you want any more changes.

**Comment by Insti** [comment]
Looks good, thanks @ajwann

**Pull request merged**